src/Sarif.Multitool/Rules/SARIF1009.IndexPropertiesMustBeConsistentWithArrays.cs

```diff
@@ -177,6 +195,6 @@ protected override void Analyze(WebResponse webResponse, string webResponsePoint
         }
 
         private static bool IndexIsValid<T>(int index, IList<T> container)
-                => index == -1 || (index >= 0 && container?.Count > index);
+                => index >= 0 && container.Count > index;
```

Comment:
>; [](start = 56, length = 1)

Yes, that's a reasonable change. If index is -1 there's nothing to worry about so you check it at the top. It's also nice because it makes the method name "true". -1 isn't a "valid" index, even though its presence means there's no rule violation.